src/main/java/edu/kit/provideq/toolbox/sat/solvers/QrispExactGroverSolver.java

```diff
@@ -0,0 +1,123 @@
+      ApplicationContext context) {
+    this.scriptPath = scriptPath;
+    this.context = context;
+  }
+
+  @Override
+  public String getName() {
+    return "Exact QRISP SAT";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Solves SAT problems using QRISP as a quantum Grover-search problem. "
+        + "It measures only possible states by utilizing result of solution numbers "
+        + "from the SharpSAT solver.";
+  }
+
+  @Override
+  public List<SubRoutineDefinition<?, ?>> getSubRoutines() {
+    return List.of(SHARPSAT_SUBROUTINE);
+  }
+
+  @Override
+  public Mono<Solution<DimacsCnfSolution>> solve(
+      String input, SubRoutineResolver subRoutineResolver, SolvingProperties properties
+  ) {
+    var solution = new Solution<>(this);
+    DimacsCnf dimacsCnf;
+    try {
+      dimacsCnf = DimacsCnf.fromString(input);
+      solution.setDebugData("Using CNF input: " + dimacsCnf);
+    } catch (ConversionException | RuntimeException e) {
+      solution.setDebugData("Parsing error: " + e.getMessage());
+      solution.abort();
+      return Mono.just(solution);
+    }
+    return subRoutineResolver.runSubRoutine(SHARPSAT_SUBROUTINE, dimacsCnf.toString())
+        .publishOn(Schedulers.boundedElastic())
+        .flatMap(sharpSatSolution -> processSharpSatResult(sharpSatSolution, dimacsCnf, solution));
+  }
+
+  private Mono<Solution<DimacsCnfSolution>> processSharpSatResult(
+      Solution<Integer> sharpSatSolution,
+      DimacsCnf dimacsCnf,
+      Solution<DimacsCnfSolution> solution
+  ) {
+    if (sharpSatSolution.getSolutionData() == null) {
+      solution.setDebugData("Sharpsat subroutine returned no solution data.");
+      solution.abort();
+      return Mono.just(solution);
+    }
+    int solutionCount = sharpSatSolution.getSolutionData();
+    solution.setDebugData("Sharpsat subroutine found " + solutionCount + " solutions.");
+    return runPythonSolver(dimacsCnf, solutionCount, solution);
+  }
+
+  private Mono<Solution<DimacsCnfSolution>> runPythonSolver(
+      DimacsCnf dimacsCnf,
+      int solutionCount,
+      Solution<DimacsCnfSolution> solution
+  ) {
+    ProcessResult<String> processResult = context
+        .getBean(PythonProcessRunner.class, scriptPath)
+        .withArguments(
+            ProcessRunner.INPUT_FILE_PATH,
+            "--solution-count", String.valueOf(solutionCount),
+            "--output-file", ProcessRunner.OUTPUT_FILE_PATH
+        )
+        .writeInputFile(dimacsCnf.toString())
+        .readOutputFile()
+        .run(getProblemType(), solution.getId());
+    if (processResult.success()) {
+      var dimacsCnfSolution = DimacsCnfSolution.fromString(
+          dimacsCnf,
+          processResult.output().orElse("")
+      );
+      solution.setSolutionData(dimacsCnfSolution);
+      solution.complete();
+    } else {
+      solution.setDebugData(processResult.errorOutput().orElse("Unknown error occurred."));
```

Comment:
Could you specify at what point in time the error occurred for a better debugging experience?